### PR TITLE
Define start and end time for segwit deployment

### PR DESCRIPTION
**Pull request by sipa**
See corresponding mailing list discussion (https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-October/013226.html) and BIP update PR (https://github.com/bitcoin/bips/pull/464).

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -93,8 +93,8 @@ class CMainParams : public CChainParams {
 
         // Deployment of SegWit (BIP141 and BIP143)
```
Comment (btcdrak) [inline src/chainparams.cpp]: also BIP147

Comment (sipa) [inline src/chainparams.cpp]: Fixed.

**Comment by btcdrak** [comment]
utACK f9c23dea976465c51b9fe2cead81b707678c16bb

**Comment by achow101** [comment]
utACK f9c23dea976465c51b9fe2cead81b707678c16bb

**Comment by JornC** [comment]
Start: http://www.wolframalpha.com/input/?i=unix+1479168000

12:00:00 am UTC | Tuesday, November 15, 2016

End: http://www.wolframalpha.com/input/?i=unix+1510704000

12:00:00 am UTC | Wednesday, November 15, 2017

utACK f9c23dea976465c51b9fe2cead81b707678c16bb

**Comment by instagibbs** [comment]
11/15/2016 @ 12:00am (UTC)
to
11/15/2017 @ 12:00am (UTC)

utACK https://github.com/bitcoin/bitcoin/pull/8937/commits/f9c23dea976465c51b9fe2cead81b707678c16bb

File: src/chainparams.cpp [hunk 2]
```diff
@@ -91,10 +91,10 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_CSV].nStartTime = 1462060800; // May 1st, 2016
         consensus.vDeployments[Consensus::DEPLOYMENT_CSV].nTimeout = 1493596800; // May 1st, 2017
 
-        // Deployment of SegWit (BIP141 and BIP143)
+        // Deployment of SegWit (BIP141, BIP143, and BIP147)
```
Comment (paveljanik) [inline src/chainparams.cpp]: No need for comma before "and".

File: src/chainparams.cpp [hunk 3]
```diff
@@ -91,10 +91,10 @@ class CMainParams : public CChainParams {
...
         consensus.vDeployments[Consensus::DEPLOYMENT_SEGWIT].bit = 1;
-        consensus.vDeployments[Consensus::DEPLOYMENT_SEGWIT].nStartTime = 0;
-        consensus.vDeployments[Consensus::DEPLOYMENT_SEGWIT].nTimeout = 0; // Never / undefined
+        consensus.vDeployments[Consensus::DEPLOYMENT_SEGWIT].nStartTime = 1479168000; // November 15th, 2016.
```
Comment (paveljanik) [inline src/chainparams.cpp]: No need for dot at the end of the line. This is the third format for simple date we use now (as shown in the Files changed in this PR) :-)

File: src/chainparams.cpp (hunk 2 above)
Comment (sipa) [inline src/chainparams.cpp]: Depends: https://en.wikipedia.org/wiki/Serial_comma

**Comment by paveljanik** [comment]
ACK /me too for simple epoch conversion ;-)

File: src/chainparams.cpp (hunk 2 above)
Comment (paveljanik) [inline src/chainparams.cpp]: Yes, I'm on the "against" side, sorry.

Comment (paveljanik) [inline src/chainparams.cpp]: BIP141, BIP143, BIP147,

C forever! ;-)

Comment (btcdrak) [inline src/chainparams.cpp]: Brit here, I approve of Oxford comma!

**Comment by TheBlueMatt** [comment]
utACK f9c23dea976465c51b9fe2cead81b707678c16bb

**Comment by jameshilliard** [comment]
utACK https://github.com/bitcoin/bitcoin/commit/f9c23dea976465c51b9fe2cead81b707678c16bb

**Comment by jl2012** [comment]
utACK f9c23de

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/8937/commits/f9c23dea976465c51b9fe2cead81b707678c16bb. Timestamps match those in the BIP update https://github.com/bitcoin/bips/pull/464/commits/5c2da7e07d924c34af4edf7098981c9a376263d2, which matches the UTC dates in the mailing list post. Let's not hold this up on comment formatting minutia.

File: src/chainparams.cpp (hunk 2 above)
Comment (Leviathn) [inline src/chainparams.cpp]: Oxford comma is best practice.

**Comment by dcousens** [comment]
utACK 8b66659

File: src/chainparams.cpp (hunk 3 above)
Comment (Xekyo) [inline src/chainparams.cpp]: Glory to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601). Should be 2016-11-15. ;)

File: src/chainparams.cpp (hunk 2 above)
Comment (rebroad) [inline src/chainparams.cpp]: honestly... does it even matter?!

**Pull request merged**